Movie detail updates

vnoelifant opened the pull request:
Refactor of the search function in views. Updates to the movie detail view and model fields. In TMDB, the videos and recommendations are different endpoints than the movie details (see https://developer.themoviedb.org/reference/intro/getting-started), but in the code, I just gave them a many to many relationship with the Movie (details) model. So you'll see that in the views.py file, in the movie_detail function, I am only passing in the movie_detail as a context, which includes videos and recommendations. I am not sure if that is  the most efficient approach, but I am able to view the data this way.  Let's discuss and review my updates at the next code review. This relates to issue #70.

bbelderbos commented on Pipfile:
```diff
@@ -13,6 +13,7 @@ django = "*"
 python-decouple = "*"
 django-bootstrap-v5 = "*"
 django-countries = "*"
+isort = "*"
```
are you using pre-commit? good way to automatically run these tools, see https://www.youtube.com/watch?v=XFyLzr5Ehf0

vnoelifant replied:
Will start

bbelderbos commented on movie_search/media_api.py:
```diff
@@ -1,15 +1,12 @@
-
-load_dotenv()
+import requests
+from django.conf import settings
```
great you moved all env variable retrieving into settings

bbelderbos commented on movie_search/media_api.py:
```diff
@@ -59,7 +56,7 @@ def get_media_data(
     url = f"{BASE_URL}{endpoint}"
 
-    params = {"api_key": API_KEY, "language": language}
+    params = {"api_key": settings.TMDB_API_KEY, "language": language}
```
better

bbelderbos commented on movie_search/models.py:
```diff
@@ -1,10 +1,9 @@
+
 class Genre(models.Model):
-    name = models.CharField(max_length=200)
+    name = models.CharField(max_length=200, null=True, blank=True, unique=True)
```
🎉 - this should solve the issue we worked on yesterday (you would have to wipe out genres table and run the fixture again, if you run the fixture again do you get a crash?)

bbelderbos commented on movie_search/models.py:
```diff
@@ -14,38 +13,51 @@ class Meta:
-    name = models.CharField(max_length=200)
-    key =  models.CharField(max_length=200)
+    name = models.CharField(max_length=200, null=True, blank=True)
+    key =  models.CharField(max_length=200, null=True, blank=True)
```
https://sentry.io/answers/django-difference-between-null-and-blank/#:~:text=Although%20we%20could%20still%20set,or%20an%20empty%20string%20%22%22%20.

<img width="790" alt="image" src="https://github.com/vnoelifant/movie-search-app/assets/387927/4d925ea0-442b-4422-aebe-e4855248bf7c">

+

<img width="721" alt="image" src="https://github.com/vnoelifant/movie-search-app/assets/387927/a0652040-9500-4da5-83a7-21217487417f">

bbelderbos commented on movie_search/models.py:
```diff
@@ -14,38 +13,51 @@ class Meta:
     revenue = models.IntegerField(default=0, null=True, blank=True)
     homepage = models.CharField(max_length=200, null=True, blank=True)
-    videos = models.ForeignKey(Video, on_delete=models.CASCADE,null=True, blank=True)
+    video = models.ManyToManyField(Video, related_name="movies", blank=True)
```
so this means that a movie can have many videos, but the videos can also be for multiple movies? it's not a 1-to-many (each video pertains to one movie only)?

vnoelifant replied:
Each trailer can only have one movie (one to many)

bbelderbos commented on movie_search/models.py:
```diff
@@ -14,38 +13,51 @@ class Meta:
     homepage = models.CharField(max_length=200, null=True, blank=True)
-    videos = models.ForeignKey(Video, on_delete=models.CASCADE,null=True, blank=True)
+    video = models.ManyToManyField(Video, related_name="movies", blank=True)
+    recommendation = models.ManyToManyField(Recommendation, related_name="movies", blank=True)
```
can you explain this relation to me please?

Similar to previous line / comment ... M2M (many-to-many) relation so: movies can have many recommendations, but each recommendation can also be for multiple movies?

vnoelifant replied:
a movie can have many recommendations, but one recommendation can apply to many movies (this will stay as a M2m relationship).

bbelderbos replied:
Are you sure? Usually a recommendation is for one movie only right? Do you have an example of this please?

bbelderbos reviewed (commented):
Good stuff, couple of questions, can you reply please?

vnoelifant commented:
Added model improvements, fixed issue #70, and GPT function refactoring. Eventually, per our Circle convo, I want to move to class refactoring. It looks like the videos side scroll section needs some adjustments because each video seems too close together. This is an example for the Mission Impossible movie from home page. @bbelderbos . 
![image](https://github.com/vnoelifant/movie-search-app/assets/43191157/776fe207-b937-4d5b-a50e-ecfcd5c44d94)

bbelderbos commented:
> It looks like the videos side scroll section needs some adjustments because each video seems too close together. This is an example for the Mission Impossible movie from home page.

ok good progress, maybe you can start with these css stylings? let me know if that helps / if you need more help ...

<img width="741" alt="image" src="https://github.com/vnoelifant/movie-search-app/assets/387927/bd3882a9-8135-4a23-affd-2cd5e8baa7cf">
